server/controllers/finance/vouchers.js

```diff
@@ -0,0 +1,134 @@
+
+/** Get list of vouchers */
+exports.list = getList;
+
+/** Get detail of vouchers */
+exports.detail = getDetail;
+
+/** Create a new voucher record */
+exports.create = create;
+
+/**
+* GET /vouchers
+*
+* @method getList
+*/
+function getList(req, res, next) {
+  var query =
+    'SELECT v.uuid, v.date, v.project_id, v.reference, v.currency_id, v.amount, ' +
+    'v.description, v.document_uuid, v.user_id, vi.uuid AS voucher_item_uuid, ' +
+    'vi.account_id, vi.debit, vi.credit ' +
+    'FROM voucher v JOIN voucher_item vi ON vi.voucher_uuid = v.uuid ';
+
+  var builder = queryCondition(query, req.query);
+
+  db.exec(builder.query, builder.conditions)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+* GET /vouchers/:uuid
+*
+* @method getDetail
+*/
+function getDetail(req, res, next) {
+  var query =
+    'SELECT v.uuid, v.date, v.project_id, v.reference, v.currency_id, v.amount, ' +
+    'v.description, v.document_uuid, v.user_id, vi.uuid AS voucher_item_uuid, ' +
+    'vi.account_id, vi.debit, vi.credit ' +
+    'FROM voucher v JOIN voucher_item vi ON vi.voucher_uuid = v.uuid ' +
+    'WHERE v.uuid = ? ;';
+  
+  db.exec(query, req.params.uuid)
+  .then(function (rows) {
+    if (!rows.length) { return next(new req.codes.ERR_NOT_FOUND()); }
+    res.status(200).json(rows[0]);
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+* POST /vouchers
+*
+* @method create
+*/
+function create(req, res, next) {
+
+  if (!checkForMissing(req.body)) { return next(new req.codes.ERR_MISSING_REQUIRED_PARAMETERS()); }
+
+  /** Initialise the transaction handler */
+  var transaction = db.transaction();
+
+  transaction.addQuery('INSERT INTO voucher SET ?', [req.body.voucher]);
+
+  transaction.addQuery('INSERT INTO voucher_item SET ?', [req.body.voucher_item]);
+
+  transaction.execute()
+  .then(function (rows) {
+    res.status(201).json({ id: req.body.voucher.uuid });
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+* @function checkForMissing
+* @param {object} The sended object
+* @return {boolean}
+* @description This functin check missing data for voucher and voucher item
+*/
+function checkForMissing(_data) {
+  var voucher = _data.voucher;
+  var voucher_item = _data.voucher_item;
+
+  var validVoucher =
+    voucher.uuid && voucher.date && voucher.project_id && voucher.reference &&
+    voucher.currency_id && (voucher.amount + 1) && voucher.description &&
+    voucher.document_uuid && voucher.user_id ;
+
+  var validVoucherItem =
+    voucher_item.uuid && voucher_item.account_id &&
+    (voucher_item.debit + 1) && (voucher_item.credit + 1) && voucher_item.voucher_uuid;
+
+  return (validVoucher && validVoucherItem);
+}
+
+/**
+* @function queryCondition
+* @param {string} _query The sql query
+* @param {object} _requestQuery The req.query object
+* @return {array} The final query and conditions
+* @description build query string conditions
+*/
+function queryCondition(_query, _requestQuery) {
```

Comment:
If you have used this in multiple controllers, you should move this out into a library.